model/riscv_pmp_control.sail

```diff
@@ -106,14 +106,42 @@ function pmpCheckRWX(ent, acc) = {
   }
 }
 
-// this needs to be called with the effective current privilege.
-val pmpCheckPerms: (Pmpcfg_ent, AccessType(ext_access_type), Privilege) -> bool
+/* this needs to be called with the effective current privilege */
+val pmpCheckPerms: (Pmpcfg_ent, AccessType(ext_access_type), Privilege) -> bool effect {rreg}
 function pmpCheckPerms(ent, acc, priv) = {
-  match priv {
-    Machine => if   pmpLocked(ent)
-               then pmpCheckRWX(ent, acc)
-               else true,
-    _       => pmpCheckRWX(ent, acc)
+  match mseccfg.MML() {
+    0b1 =>
+      match (ent.R(), ent.W(), ent.L(), ent.X()) {
+        (0b0, 0b1, 0b1, _) =>
+          match acc {
+            Execute(_) => true,
+            Read(_)    => priv == Machine & ent.X() == 0b1,
+            _          => false
+          },
+        (0b0, 0b1, 0b0, _) =>
+          match acc {
+            Read(_)  => true,
+            Write(_) => priv == Machine | ent.X() == 0b1,
+            _        => false
+          },
+        (0b1, 0b1, 0b1, 0b1) =>
+          match acc {
+            Read(_) => true,
```

Comment:
This should be an inaccessible region right?

Reply:
No, It should not be an inaccessible region. Spec says

> The encoding pmpcfg.LRWX=1111 can be used for sharing data between M-mode and S/U mode,
where both modes only have read-only access to the region. The rule remains locked so that any
further modifications to its associated configuration or address registers are ignored until a PMP
reset, unless mseccfg.RLB is set.

My implementation is according to the spec.